### PR TITLE
performs ~pritive black VoOdOo base64 MaGiX~ with the ulterior hope of eliminating PDA lag!!

### PR DESCRIPTION
**Pull request by 9600bauds**
Here is what I THINK was happening:
When you open your PDA for the first time the serb has to send you all the icons. The icons on themselves are like 200 bytes, so that shouldn't take a long time. The problem is that each icon is its own request, so 20 small files take a much longer time to download than 1 file that is 50 times bigger than all those.

What this does is move all the PDA icons from their own files to base64 data URIs. Essentially, all the icons are stored in plaintext in one big stylesheet.

IF NOTHING GOES WRONG, and IF THAT WAS THE CAUSE OF THE SLOWDOWN, theoretically there shouldn't be any slowdown anymore after #13352 and this together.

Oh and also added the whole list-jointext-mumbojumbo from the R&D console for a marginal performance improvement on these things

:cl:
 * experiment: Did some black magic on PDAs to try to eliminate the long delay when opening them for the first time in a round. If your PDA's icons randomly go missing, please make a bug report saying what happened!

### CODE REVIEW

**Comment by optimumtact** [comment]
I swear to god byond's failure to bundle it's assets is going to make us do this for everything one day

File: code/game/objects/items/devices/PDA/PDA.dm [hunk 1]
```diff
@@ -622,27 +622,32 @@ var/global/list/obj/item/device/pda/PDAs = list()
...
+	if(user.client)
+		var/datum/asset/simple/C = new/datum/asset/simple/pda()
+		send_asset_list(user.client, C.assets)
+
+	var/dat = list()
+	dat += {"
+	<html>
+	<head><title>Personal Data Assistant</title></head>
+	<body>
+	<link rel="stylesheet" type="text/css" href="pda.css"/> <!--This stylesheet contains all the PDA icons in base 64!-->
```
Comment (9600bauds) [inline code/game/objects/items/devices/PDA/PDA.dm:634]: Yes, this is rather primitive sadly. I know HTML datums handle this stuff by themselves, but I kept running into the same inscrutable problems with stylesheets hating my soul.

**Comment by 9600bauds** [comment]
thank lumexx

**Comment by 9600bauds** [comment]
Here is the shitty Python script I used to convert all the icons automagically (DONUT COPY OR I DMCMAM YOU!!) (jk):
```
import base64
import PIL
from PIL import Image
import os, sys

css_file = open("Output.txt", "w")
path = "D:\Desktop\PDASHITS"
dirs = os.listdir( path )


for filename in os.listdir(os.getcwd()):
    if not filename.endswith(".png"):
        continue
    print(filename)
    css_file.write(".")
    css_file.write(filename.replace('.png',''))
    css_file.write("\n{")
    css_file.write("background-image: url(data:image/png;base64,")
    css_file.write(str(base64.b64encode(open(filename, "rb").read()).decode('ascii')))
    css_file.write(");}\n")

css_file.close()
```

and a website to do this too http://websemantics.co.uk/online_tools/image_to_data_uri_convertor

**Comment by PJB3005** [comment]
Basically, I believe because BYOND has a browse queue, the files arrive at the client in the order they're sent.

Now, BYOND also does checksum checking on the files to prevent sending the file if the client doesn't have it yet.

HILARIOUSLY ENOUGH, this is often *slower* than flat out giving it the file.

So on top of having hideous tons of RTT because of the queue, it somehow manages to fuck up even basic check-summing and sending of the files.

Thanks BYOND.

It would literally be easier to just find an IE8 exploit to abuse jesus.

File: code/modules/client/global cache.dm [hunk 2]
```diff
@@ -152,42 +152,7 @@
...
-		"pda_scanner.png"		= 'icons/pda_icons/pda_scanner.png',
-		"pda_signaler.png"		= 'icons/pda_icons/pda_signaler.png',
-		"pda_status.png"		= 'icons/pda_icons/pda_status.png',
-		"pda_clock.png"			= 'icons/pda_icons/pda_clock.png',
-		"pda_game.png"		= 'icons/pda_icons/pda_game.png',
-		"pda_egg.png"			= 'icons/pda_icons/pda_egg.png',
-		"pda_money.png"				= 'icons/pda_icons/pda_money.png',
-		"pda_minimap_box.png"	= 'icons/pda_icons/pda_minimap_box.png',
-		"pda_minimap_bg_notfound.png"	= 'icons/pda_icons/pda_minimap_bg_notfound.png',
-		"pda_minimap_deff.png"					= 'icons/pda_icons/pda_minimap_deff.png',
```
Comment (9600bauds) [inline code/modules/client/global cache.dm]: Oh yeah do not merge this until after #13352

Comment (Probe1) [inline code/modules/client/global cache.dm]: GET TO WORK CODEMONKEY

**Comment by kol1th** [comment]
Pray to the gods. 🙏 📿

**Comment by Exxion** [comment]
Fixes #6910, but Probe already closed that
What's with all these issues getting fixed right AFTER Probe closes them?

**Comment by Probe1** [comment]
Meme magic

**Comment by Probe1** [comment]
🙏

**Comment by Probe1** [comment]
Also I tested it and it worked as intended with no runtimes.

**Comment by 9600bauds** [comment]
clip 2014-01-23 at 04 51 26

**Comment by Intigracy** [comment]
They get fixed after probe closes them because someone sees probe close it, says "hey I can fix that", and fixes it.

**Comment by Probe1** [comment]
That would be the proper real answer.

**Comment by 9600bauds** [comment]
Oh look it worked

**Comment by clusterfack** [comment]
I told you guys you could do this how many months ago?

**Comment by 9600bauds** [comment]
I wasn't there when you said that

**Comment by PJB3005** [comment]
Just like my ability to care.

**Pull request merged**